SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Channel/Channel.cs

```diff
@@ -104,37 +109,30 @@ public bool IsEnabled
         public event EventHandler<FailedToSendLogEventArgs> FailedToSendLog;
         #endregion
 
-        public Task Enqueue(Log log)
+        public async Task EnqueueAsync(Log log)
         {
-            _mutex.Lock();
-            var stateSnapshot = _stateKeeper.GetStateSnapshot();
             try
             {
+                await _mutex.LockAsync().ConfigureAwait(false);
+                var stateSnapshot = _stateKeeper.GetStateSnapshot();
                 if (_discardLogs)
                 {
                     MobileCenterLog.Warn(MobileCenterLog.LogTag, "Channel is disabled; logs are discarded");
                     _mutex.Unlock();
                     SendingLog?.Invoke(this, new SendingLogEventArgs(log));
                     FailedToSendLog?.Invoke(this, new FailedToSendLogEventArgs(log, new CancellationException()));
-                    return null;
                 }
                 _mutex.Unlock();
                 EnqueuingLog?.Invoke(this, new EnqueuingLogEventArgs(log));
-                _mutex.Lock(stateSnapshot);
-                return Task.Run(async () =>
+                await Task.Run(async () =>
                 {
                     await PrepareLogAsync(log).ConfigureAwait(false);
```

Comment:
This probably should have been done before, but PrepareLogAsync should take a statesnapshot parameter, and lock around the setting of _device. But the call to GetDeviceInformationAsync should be done before the lock is acquired